NeuroID/src/main/java/com/neuroid/tracker/service/NIDJobServiceManager.kt

```diff
@@ -155,11 +157,55 @@ class NIDJobServiceManager(
                             logger.e(msg = "network failure, sendEventsNow() failed retrylimitHit: ${!isRetry} $message")
                         }
                     },
-                    dataStore
                 )
             } ?: run {
                 userActive = false
             }
         }
     }
+
+
+    /**
+     * Get the current system memory state.
+     */
+    private fun checkMemoryLevel(context: Context): NIDEventModel? {
+        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
```

Comment:
The activityManager var being allocated here is being assigned this system level singelton every time the method is called (using more memory). This should be probably be allocated once at the begining of the file.

Reply:
Good catch, init with start job but to make doubly sure that we won't break I put in a backup use case. This can be refactored when we turn the jobServiceManager into a proper class with an init